filebeat/prospector/prospector.go

```diff
@@ -214,12 +223,20 @@ func (p *Prospector) updateState(event *input.Event) error {
 	event.Module = p.config.Module
 	event.Fileset = p.config.Fileset
 
-	ok := p.outlet.OnEvent(event)
-	if !ok {
-		logp.Info("Prospector outlet closed")
-		return errors.New("prospector outlet closed")
+	eventMap := event.ToMapStr()
+
+	//run the filters before sending to
+	eventMap = p.filters.Run(eventMap)
+	if eventMap != nil {
```

Comment:
Assuming the event was "removed" by the filter, this would probably mean no state update is sent anymore? Or when exactly is `eventMap` empty?